pkg/featflag/features_test.go

```diff
@@ -17,71 +17,248 @@
+		subject.ChefFeatAll.Equals(&featFlag),
+		"the CHEF_FEAT_ALL feature flags should never change",
+	)
+
+	// the same goes for the global analyze feature flag
+	featFlag = subject.New("MY_ANALYZE_FEATURE", "analyze")
+	assert.Equal(t, "CHEF_FEAT_ANALYZE", featFlag.Env())
+	assert.Equal(t, "analyze", featFlag.Key())
+	assert.True(t,
+		subject.ChefFeatAnalyze.Equals(&featFlag),
+		"the CHEF_FEAT_ANALYZE feature flags should never change",
+	)
+}
+
+//
+// tests that involve modifying the config (config.toml)
+//
+func TestAccessingFlagsWithNilConfig(t *testing.T) {
+	subject.LoadConfig(nil)
+	assert.False(t, subject.ChefFeatAll.Enabled())
+	assert.False(t, subject.ChefFeatAnalyze.Enabled())
+}
+
+func TestEnableFlagsFromConfigToml(t *testing.T) {
+	// all flags should be disabled
+	subject.LoadConfig(nil)
+	assert.False(t,
+		subject.ChefFeatAll.Enabled(),
+		"global ALL feature flag should be disabled",
+	)
+	assert.False(t,
+		subject.ChefFeatAnalyze.Enabled(),
+		"global ANALYZE feature flag should be disabled",
+	)
+	// even new flags created locally
+	var chefFeatXYZ = subject.New("CHEF_FEAT_XYZ", "xyz")
+	assert.False(t,
+		chefFeatXYZ.Enabled(),
+		"local XYZ feature flag should be disabled",
+	)
+
+	// initializing a configuration file
+	createConfigToml(t)
+	defer os.RemoveAll(".chef-workstation") // clean up
+
+	// loading the config to the featflag library
+	customConfig, err := config.New()
+	if err != nil {
+		t.Fatal("unable to load config.toml, check why this happened", err.Error())
+	}
+	subject.LoadConfig(&customConfig)
+
+	// now things start getting interesting
+	assert.False(t,
+		subject.ChefFeatAll.Enabled(),
+		"global ALL feature flag should continue to be disabled",
+	)
+	assert.True(t, // now it is enabled! (from the config)
+		subject.ChefFeatAnalyze.Enabled(),
+		"global ANALYZE feature flag should be enabled from loaded config.toml",
+	)
+	assert.False(t,
+		chefFeatXYZ.Enabled(),
+		"local XYZ feature flag should continue to be disabled",
+	)
 
-	assert.Equal(t, "", chefFeatUnknown.Name())
-	assert.False(t, chefFeatUnknown.Enabled())
+	// since environment variables has more priorityby, by enabling the
+	// global feature flag CHEF_FEAT_ALL all features should be enabled
+	defer os.Setenv("CHEF_FEAT_ALL", "")
+	if err := os.Setenv("CHEF_FEAT_ALL", "gimmeall"); err != nil {
+		t.Fatal("unable to set environment variable")
+	}
+
+	assert.True(t,
+		subject.ChefFeatAll.Enabled(),
+		"global ALL feature flag should be now enabled",
+	)
+	assert.True(t, // now it is enabled! (from the config)
+		subject.ChefFeatAnalyze.Enabled(),
+		"global ANALYZE feature flag should continue to be enabled",
+	)
+	assert.True(t,
+		chefFeatXYZ.Enabled(),
+		"local XYZ feature flag should be now enabled",
+	)
+}
+
+// when calling this function, make sure to add the defer clean up as the below example
+// ```go
+// createConfigToml(t)
+// defer os.RemoveAll(".chef-workstation") // clean up
+// ```
+func createConfigToml(t *testing.T) {
```

Comment:
Same comment here as another review we looked at last week:

1. Lets make this return the directory that is created so it can be called `defer os.RemoveAll(createConfigToml(t))`
2. Can we have all this testing occur in a temporary folder so we don't accidently leave around a `.chef-workstation` folder in someone's working directory?

Happy to pair on these changes so I can learn the correct way to implement this pattern.

Reply:
Yeah, we could tackle https://github.com/chef/chef-analyze/issues/57 today! 💯